pkg/extensions/search/userprefs_test.go

```diff
@@ -806,6 +806,97 @@ func TestGlobalSearchWithUserPrefFiltering(t *testing.T) {
 	})
 }
 
+func TestExpandedRepoInfoWithUserPrefs(t *testing.T) {
+	Convey("ExpandedRepoInfo with User Prefs", t, func() {
```

Comment:
Maybe there should be also a repo which is not starred/bookmarked, and we test that as well.

Reply:
I have added exhaustive tests for all cases